pkg/scheduler/schedule_one.go

```diff
@@ -115,6 +122,9 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		if !status.IsSuccess() {
 			sched.handleBindingCycleError(bindingCycleCtx, state, fwk, assumedPodInfo, start, scheduleResult, status)
```

Comment:
is there an error case where we don't call `AddUnschedulableIfNotPresent`?

Reply:
Here, in handleBindingCycleError, we 100% call `sched.FailureHandler` ([ref](https://github.com/sanposhiho/kubernetes/blob/6e09e4e7d68bbf63540f6289a48864c7f2f6846f/pkg/scheduler/schedule_one.go#L317)) and `sched.FailureHandler` will call `AddUnschedulableIfNotPresent`.

And, in `sched.FailureHandler` (actually, sched.handleSchedulingFailure), `AddUnschedulableIfNotPresent` won't be called if:
- `podLister.Pods(pod.Namespace).Get(pod.Name)` returns error
  - https://github.com/sanposhiho/kubernetes/blob/6e09e4e7d68bbf63540f6289a48864c7f2f6846f/pkg/scheduler/schedule_one.go#L960
  - 
- `len(cachedPod.Spec.NodeName) != 0`
  - https://github.com/sanposhiho/kubernetes/blob/6e09e4e7d68bbf63540f6289a48864c7f2f6846f/pkg/scheduler/schedule_one.go#L966


To prevent memory leak via these scenarios, I added this change: [bffd0f8](https://github.com/kubernetes/kubernetes/pull/118438/commits/bffd0f821cd644d6ca2ed943b47189171fe38efe)